src/pages/Overview/BalanceChart.tsx

```diff
@@ -0,0 +1,283 @@
+  const totalBalance = planckToUnit(
+    free.plus(poolBondOpions.active).plus(unlockingPools),
+    units
+  );
+  // convert balance to fiat value
+  const freeFiat = totalBalance.multipliedBy(
+    new BigNumber(prices.lastPrice).decimalPlaces(2)
+  );
+
+  // total funds nominating
+  const nominating = planckToUnit(
+    allTransferOptions.nominate.active
+      .plus(allTransferOptions.nominate.totalUnlocking)
+      .plus(allTransferOptions.nominate.totalUnlocked),
+    units
+  );
+  // total funds in pool
+  const inPool = planckToUnit(
+    allTransferOptions.pool.active
+      .plus(allTransferOptions.pool.totalUnlocking)
+      .plus(allTransferOptions.pool.totalUnlocked),
+    units
+  );
+
+  // check account non-staking locks
+  const locks = getLocks(activeAccount);
+  const locksStaking = locks.find(({ id }) => id === 'staking');
+  const lockStakingAmount = locksStaking
+    ? locksStaking.amount
+    : new BigNumber(0);
+
+  // total funds available, including existential deposit, minus staking.
+  const graphAvailable = planckToUnit(free.minus(lockStakingAmount), units);
+  const notStaking = graphAvailable;
+
+  // graph percentages
+  const graphTotal = nominating.plus(inPool).plus(graphAvailable);
+  const graphNominating = greaterThanZero(nominating)
+    ? nominating.dividedBy(graphTotal.multipliedBy(0.01))
+    : new BigNumber(0);
+
+  const graphInPool = greaterThanZero(inPool)
+    ? inPool.dividedBy(graphTotal.multipliedBy(0.01))
+    : new BigNumber(0);
+
+  const graphNotStaking = greaterThanZero(graphTotal)
+    ? new BigNumber(100).minus(graphNominating).minus(graphInPool)
+    : new BigNumber(0);
+
+  // available balance data
+  const fundsLocked = planckToUnit(frozen.minus(lockStakingAmount), units);
+  let fundsReserved = planckToUnit(edReserved.plus(feeReserve), units);
+  const fundsFree = planckToUnit(
+    BigNumber.max(allTransferOptions.freeBalance.minus(feeReserve), 0),
+    units
+  ).minus(fundsLocked);
+  // available balance percentages
+  const graphLocked = greaterThanZero(fundsLocked)
+    ? fundsLocked.dividedBy(graphAvailable.multipliedBy(0.01))
+    : new BigNumber(0);
+
+  const graphFree = greaterThanZero(fundsFree)
+    ? fundsFree.dividedBy(graphAvailable.multipliedBy(0.01))
+    : new BigNumber(0);
+
+  // get total available balance, including reserve and locks
+  if (graphAvailable.isLessThan(fundsReserved)) {
+    fundsReserved = graphAvailable;
+  }
+
+  // formatter for price feed.
+  const usdFormatter = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD',
+  });
+
+  const isNominating = greaterThanZero(nominating);
+  const isInPool = greaterThanZero(
+    poolBondOpions.active
+      .plus(poolBondOpions.totalUnlocked)
+      .plus(poolBondOpions.totalUnlocking)
+  );
+
+  return (
+    <>
+      <CardHeaderWrapper>
+        <h4>{t('overview.balance')}</h4>
+        <h2>
+          <Token className="networkIcon" />
+          <Odometer
```

Comment:
This widget seem to have problems with displaying big balances. When you connect to Devnet, and import account `//0` (ie Validator 0). then you see

![image](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/assets/3909333/8f59a379-8bd7-487f-85ef-26ea1519dbd5)

Whereas you'd expect some non-zero number. For comparison how it looks on azero.dev

![image](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/assets/3909333/092d178d-d497-44d9-b879-94473971ee2e)

Reply:
I can see this is now fixed. Thanks!

Reply:
Actually, `//1` account still has this problem:

![image](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/assets/3909333/5d3156f0-8860-42cb-b0e2-5e5286552e0f)

Reply:
So the bug seems to be with method `minDecimalPlaces`. So on the widget, this is used
```
<Odometer
            value={minDecimalPlaces(totalBalance.toFormat(), 2)}
            zeroDecimals={2}
          />
```
Now `totalBalance.toFormat()` is correct and logged in console as `20,000,000`. However the result of `minDecimalPlaces(20,000,000, 2)` is `NaN`.

Reply:
`minDecimalPlaces` source code comes from here https://www.npmjs.com/package/@polkadot-cloud/utils/v/0.0.11?activeTab=code

Reply:
Btw on `main`, they use most recent version of this package which is `0.0.22`: https://github.com/paritytech/polkadot-staking-dashboard/blob/main/yarn.lock#L610

Reply:
It turned out `yarn dev --force` was needed as patches are resolved only on install phase.